libs/filer/workspace_fuse_client.go

```diff
@@ -0,0 +1,218 @@
+package filer
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"io/fs"
+	"path"
+	"slices"
+	"strings"
+
+	"github.com/databricks/cli/libs/notebook"
+	"github.com/databricks/databricks-sdk-go"
+	"github.com/databricks/databricks-sdk-go/service/workspace"
+	"golang.org/x/exp/maps"
+)
+
+type workspaceFuseClient struct {
+	workspaceFilesClient Filer
+}
+
+func stripNotebookExtension(ctx context.Context, w Filer, name string) (stripName string, stat fs.FileInfo, ok bool) {
+	ext := path.Ext(name)
+
+	extensionsToLanguages := map[string]workspace.Language{
+		".py":    workspace.LanguagePython,
+		".r":     workspace.LanguageR,
+		".scala": workspace.LanguageScala,
+		".sql":   workspace.LanguageSql,
+		".ipynb": workspace.LanguagePython,
+	}
```

Comment:
This is constant, can be lifted outside this function.

Reply:
Done